src/container/PokeContainer.jsx

```diff
@@ -20,22 +22,72 @@ function PokeContainer(){
     pokemonData();
   }, []);
 
+  //HANDLE SUBMIT
   const handleSubmit = async (event) =>{
     event.preventDefault();    
     setIsLoading(true);
 
-    const pokemonData = await fetchPokemonBySearch();
+    const pokemonData = await fetchPokemonBySearch(searchName);
     setPokemons(pokemonData);
+   
     setIsLoading(false);
     setSearchName('');
   };
 
+  //SORT POKEMONS
+  const handleSort = async (event) => {
+    setIsLoading(true);
+    setSorted(event.target.value);
+    const sortedPokes = await fetchSortedPokemon(sorted);
+    await setPokemons(sortedPokes);
+    // setSorted('');
+    setIsLoading(false);
+  };  
+
+
+//FETCH ABILITIES
+  useEffect(() => {
+    const abily = async () => {
+      const fetchedAbilities = await fetchPokemonAbilities();
+      setAbilities(fetchedAbilities);
+    };
+    
+    abily();
+
+  }, []);
+
+
+  //FILTERED POKEMON
+  useEffect(()=> {
+    async function capturedFilteredPokemon(){
+      if (!selectedAbility) return;
+      setIsLoading(true);
+
+      if (selectedAbility !== 'all'){
+        const filteredPokemon = await fetchFilteredPokemon(selectedAbility);
+       
+        setPokemons(filteredPokemon);
+      }
+      setIsLoading(false);
+    }
+    capturedFilteredPokemon();
+  }, [selectedAbility]);
+
+
+
+
   return (
     <main>
       <Controls
         searchName= {searchName}
         setSearchName = {setSearchName}
         handleSubmit={handleSubmit}
+        abilities={abilities}
+        setSelectedAbility={setSelectedAbility}
+        selectedAbility={selectedAbility}
+        sorted={sorted}
+        setSorted={setSorted}
+        handleSort={handleSort}
       />
       <div>
         {isloading ? (<h1> Loading Pokemons!!!!</h1>)
```

Comment:
You can also put this loading conditional above the controls if you want it to cover the whole page.